### PR TITLE
completed week-5-assessment

### PR DESCRIPTION
**Pull request by LoneWanderer101**
# Submitting Week 5 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,162 @@
...
-// Expected output: "3cc3ntr1c"
-
+// const secretCodeWord1 = "Lackadaisical"
+// // Expected output: "L4ck4d41s1c4l"
+// const secretCodeWord2 = "Gobbledygook"
+// // Expected output: "G0bbl3dyg00k"
+// const secretCodeWord3 = "Eccentric"
+// // Expected output: "3cc3ntr1c"
+
+describe ("message", () => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:25]: nice test

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,162 @@
...
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+    // Create a function called "message" that takes in a string
+    // return a new string whith a coded message that replaced 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+    // i tried the code /a/, 4 however, it did not pass. After conducting A LOT of reserach i found the following... The gi modifier is used to do a **case insensitive search** of all occurrences of a regular expression in a string. This worked because it's case INsensitive.
+
+    const message = (string) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:42]: awesome one-liner return

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,162 @@
...
+        const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+        const letterE = "e"
+        expect(partLetter(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+        expect(partLetter(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+// Create a function called partLetter that takes in an array of words and a single letter and returns all the words that contain that particular letter.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:81]: like your pseudocode

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,42 +15,162 @@
...
+// it will take two arguments; the array and the particular letter
+// use .filter on the array which creates a new array filled with elements that pass a test provided by a function.
+// since the array includes the letters provided as capital letters, you have to turn all capital letters to lower case in order to match with the lower case letters provided ('a', 'e')
+// then return the value and the new array which should output the new array with the correct words
+
+
+const partLetter = (arr, LetterA) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:88]: great

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,42 +15,162 @@
...
-// Expected output: false
-
+// const hand1 = [5, 5, 5, 3, 3]
+// // Expected output: true
+// const hand2 = [5, 5, 3, 3, 4]
+// // Expected output: false
+// const hand3 = [5, 5, 5, 5, 4]
+// // Expected output: false
+
+describe ("fullHouse", () => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:119]: indentation looks good

File: code-challenges.test.js [hunk 6]
```diff
@@ -15,42 +15,162 @@
...
+
+        // const hand1 = [5, 5, 5, 3, 3]
+        // const hand2 = [5, 5, 3, 3, 4]
+        // const hand3 = [5, 5, 5, 5, 4]
+        
+        // const hand4 = ['K', 'K', 'K', 'Q', 'Q']
+        // const hand5 = [2, 2, 3, 'A', 'A']
+        // tried the code with aces, queen, and king cards and it worked.
+
+const fullHouse = (arr) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:156]: terrific

File: interview-questions.md [hunk 7]
```diff
@@ -8,52 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a hash in Ruby?
 
   Your answer:
+  A hash in ruby is a collection of key:value pairs. 
 
   Researched answer:
-
+  A hash in ruby is a collection of key:value pairs. A hash is similar to an array except taht indexing is done via arbitrary keys of an object type and not trough integer index.If you attempt to access a hash with a key that does not exist, the method will return nil.
```
Comment (SunkissedQueen) [inline interview-questions.md]: What is an example that you can describe? Preferably something you have done

File: interview-questions.md [hunk 8]
```diff
@@ -8,52 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 3. What is Ruby on Rails?
 
   Your answer:
+  Ruby on Rails is a web application framework written in Ruby. Rails is a MVC framework.
 
   Researched answer:
-
-
+  Rails is a model–view–controller (MVC) framework, providing default structures for a database, a web service, and web pages. Rails facilitates the use of web standards such as JSON or XML for data transfer and HTML, CSS and JavaScript for user interfacing. Rails can be used to build web applications taht span both the front end and back end. The Model and Controller would be considered the back-end part of the application. The View handles the front end by generating the actual web page that people see in the browser, along with included JavaScript and HTML. You can also use Rails to create web services or APIs that return JSON for use by other applications.
```
Comment (SunkissedQueen) [inline interview-questions.md:31]: Some lines look very similar to the following article: https://www.hashlabs.com/services/ruby-on-rails/#:~:text=Rails%20is%20a%20model%E2%80%93view,for%20display%20and%20user%20interfacing

File: ruby-challenges.rb [hunk 9]
```diff
@@ -10,6 +10,15 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def find_word array, letter
+    array.select { |value| value.include? letter }
+end    
+    p find_word beverages_array, letter_o
+
+def find_word array, letter
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:18]: Why was the method created twice?

File: ruby-challenges.rb [hunk 10]
```diff
@@ -20,16 +29,90 @@
 # Expected output: 100
 
 
+def sum_array array
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:32]: Appreciate your sharing multiple ways

File: ruby-challenges.rb [hunk 11]
```diff
@@ -20,16 +29,90 @@
...
+    # output --> 10
+    my_bike.pedal_faster(18)
+    p my_bike.current_speed
+    # output--> 28
+
+    my_bike.brake(5)
+    p my_bike.current_speed
+    # output --> 23
+
+    my_bike.brake(25)
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:116]: Good work

**Review by SunkissedQueen — DISMISSED** [review]
Jorge, I am requesting a change. Please place the researched answers in your own word. Ensure you are addressing the definition, an example of a project you created involving that tool, and something interesting that you found from the research.

**Review by SunkissedQueen — CHANGES_REQUESTED** [review]
Jorge, I am requesting a change. Please place the researched answers in your own word. Some lines look very similar to the following article: https://www.hashlabs.com/services/ruby-on-rails/#:~:text=Rails%20is%20a%20model%E2%80%93view,for%20display%20and%20user%20interfacing
Ensure you are addressing the definition, an example of a project you created involving that tool, and something interesting that you found from the research.

**Review by SunkissedQueen — APPROVED** [review]
Jorge, I am confident in your coding skillset. The interviews are for you and and a great way to practice conveying how you understand the concepts. Don't rob yourself of the opportunity. Next week please make sure that you are addressing everything in your own words--->the definition, an example of a project you created involving that tool, and something interesting that you found from the research. Practice as much as you can.

**Comment by LoneWanderer101** [comment]
Thank you charlean

**Pull request merged**